### DISCUSSION TITLE
`xref-find-apropos` behaviour is different in source code buffer and in REPL buffers

### DISCUSSION THREAD

**Discussion by Andre0991** [joaotavora/eglot#984] [General]
Hi.

I'm not sure `eglot` is directly related to the issue, of if it's an issue at all, so I'm starting a discussion w/o too many details.

In a Clojure project, if I use `xref-find-apropos` in a source code buffer, it correctly uses LSP for finding stuff.
However, if I run `xref-find-apropos` in the REPL buffer associated with the same project, it asks me for tags.

I suppose `xref` is not considering `eglot` as one of its backends in the REPL buffer.
Why is that? Is this related to `project.el`?

**Comment by jdtsmith** [comment]
Lsp doesn’t consider REPL’s at all. It’s only for source code files.

**Reply by Andre0991** [reply]
That was my suspicion.

But then, my question is how to make `xref-find-apropos` work in a REPL with the LSP backend, if this is possible at all.

**Reply by jdtsmith** [reply]
What is your expectation of what find-apropos should do in a REPL?  In eglot from a buffer it sends a workspace/symbol request, and your LSP server, knowing your project/workspace, searches for that symbol and returns matches.  What is the "workspace" in a REPL?  _If_ you have one REPL per workspace, it's perhaps possible to hook it up by switching to the first buffer whose (e.g., for python) `(python-shell-get-process)` matches `(get-buffer-process (current-buffer))` in the REPL, and the completing the apropos call from there; see `xref-backend-apropos`.

**Reply by Andre0991** [reply]
Cool, I took inspiration from your suggestion and came up with this function:

```emacs-lisp
(defun apt--inf-clojure-repl-apropos
    ()
  (interactive)
  (if-let ((sym (symbol-at-point)))
    (progn
      (inf-clojure-switch-to-recent-buffer)
      (let ((pattern (read-string "Pattern: "
				  (symbol-name sym))))
	(when pattern
	  (xref-find-apropos pattern))))
    (message "No symbol at point.")))
```
